test-data/unit/check-dataclasses.test

```diff
@@ -2420,3 +2420,16 @@ class Test(Protocol):
     def reset(self) -> None:
         self.x = DEFAULT
 [builtins fixtures/dataclasses.pyi]
+
+[case testStrictOptionalAlwaysSet]
+# flags: --strict-optional
```

Comment:
```suggestion
```
`--strict-optional` is the default. So the flag wouldn't be necessary.

Reply:
`--strict-optional` is actually off by default in tests (at least for `testcheck.py`).

Reply:
Interesting, before writing the initial comment, I removed the flag from the test case and it still passed 🤔

> `--strict-optional` is actually off by default in tests (at least for `testcheck.py`).

I enabled it for pythoneval tests in #15474, seems I should work on the rest as well.

Reply:
> Interesting, before writing the initial comment, I removed the flag from the test case and it still passed

This is exactly why I moved it to pythoneval, otherwise it doesn't really test much.